File: nagasena/src/org/openexi/proc/events/EXIEventComment.java
package org.openexi.proc.events;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.schema.Characters;

public final class EXIEventComment implements EventDescription {

  private final Characters m_text;
  private final EventType m_eventType;

  public EXIEventComment(Characters text, EventType eventType) {
    assert eventType.itemType == EventType.ITEM_CM;
    m_text = text;
    m_eventType = eventType;
  }

  ///////////////////////////////////////////////////////////////////////////
  // Implementation of EXIEvent interface
  ///////////////////////////////////////////////////////////////////////////

  public byte getEventKind() {
    return EventDescription.EVENT_CM;
  }
  
  public String getURI() {
    return null;
  }
  
  public String getName() {
    return null;
  }

  public int getURIId() {
    return -1;
  }
  
  public int getNameId() {
    return -1;
  }

  public String getPrefix() {
    return null;
  }

  public Characters getCharacters() {
    return m_text;
  }

  public BinaryDataSource getBinaryDataSource() {
    return null;
  }

  public final EventType getEventType() {
    return m_eventType;
  }

}

File: nagasenacs/src/Nagasena/Proc/Events/EXIEventComment.cs
using System.Diagnostics;

using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.Events {

  /// <exclude/>
  public sealed class EXIEventComment : EventDescription {

    private readonly Characters m_text;
    private readonly EventType m_eventType;

    public EXIEventComment(Characters text, EventType eventType) {
      Debug.Assert(eventType.itemType == EventType.ITEM_CM);
      m_text = text;
      m_eventType = eventType;
    }

    ///////////////////////////////////////////////////////////////////////////
    // Implementation of EXIEvent interface
    ///////////////////////////////////////////////////////////////////////////

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_CM;
      }
    }

    public string URI {
      get {
        return null;
      }
    }

    public string Name {
      get {
        return null;
      }
    }

    public int URIId {
      get {
        return -1;
      }
    }

    public int NameId {
      get {
        return -1;
      }
    }

    public string Prefix {
      get {
        return null;
      }
    }

    public Characters Characters {
      get {
        return m_text;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

    public EventType getEventType() {
      return m_eventType;
    }

  }

}